src/execute/modify_contract.rs

```diff
@@ -0,0 +1,96 @@
+use crate::ask_order::ASKS_V1;
+use crate::bid_order::BIDS_V3;
+use crate::common::ContractAction;
+use crate::contract_info::{get_contract_info, modify_contract_info};
+use crate::error::ContractError;
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+use provwasm_std::{ProvenanceMsg, ProvenanceQuery};
+use std::collections::HashSet;
+
+pub fn modify_contract(
+    deps: DepsMut<ProvenanceQuery>,
+    _env: Env,
+    info: &MessageInfo,
+    approvers: Option<Vec<String>>,
+    executors: Option<Vec<String>>,
+    ask_fee_rate: Option<String>,
+    ask_fee_account: Option<String>,
+    bid_fee_rate: Option<String>,
+    bid_fee_account: Option<String>,
+    ask_required_attributes: Option<Vec<String>>,
+    bid_required_attributes: Option<Vec<String>>,
+) -> Result<Response<ProvenanceMsg>, ContractError> {
+    let contract_info = get_contract_info(deps.storage)?;
+
+    if !contract_info.executors.contains(&info.sender) {
+        return Err(ContractError::Unauthorized);
+    }
+
+    let contains_ask = !ASKS_V1.is_empty(deps.storage);
+    match &ask_required_attributes {
+        None => (),
+        Some(_) => {
+            if contains_ask {
+                return Err(ContractError::InvalidFields {
+                    fields: vec!["ask_required_attributes".to_string()],
+                });
+            }
+        }
+    }
+
+    let contains_bid = !BIDS_V3.is_empty(deps.storage);
+    if contains_bid {
+        match &bid_required_attributes {
+            None => {}
+            Some(_) => {
+                return Err(ContractError::InvalidFields {
+                    fields: vec!["bid_required_attributes".to_string()],
+                });
+            }
+        }
+        match (&bid_fee_rate, &bid_fee_account) {
```

Comment:
should we be doing the same check for ask fee / ask account when checking if there are any ask orders

Reply:
i think this is related to how fees are taken.

Bids:
- Fees are sent with the bid order creation
- If the fee changes on the bid side, more/less fees would need to be accounted for the existing bid orders

Asks:
- Fees are withheld during match settlement
- if the ask fee changes, it will just change how much is withheld after the fact

Reply:
ya, it is outside the scope of this change, and technically it isn't an issue, but if an order is placed and the fees are 1% and then they are changed to 50%, the seller would be upset.